projects/phone/src/lib/masks/phone/processors/validate-phone-preprocessor.ts

```diff
@@ -23,29 +24,39 @@ export function validatePhonePreprocessorGenerator({
 
         // handling autocomplete
         if (value && !value.startsWith(cleanCode) && !data) {
-            return {elementState: {value: prefix + value, selection}};
+            const formatter = new AsYouType({defaultCountry: countryIsoCode}, metadata);
+
+            formatter.input(value);
+            const numberValue = formatter.getNumberValue() || '';
+
+            formatter.reset();
+
+            return {elementState: {value: formatter.input(numberValue), selection}};
```

Comment:
Could you elaborate on this part, why do we feed it back in?

Reply:
yes, for example, if in autocomplete the phone number starts with 8 (89201234567)
first we parse the phone using the AsYouType formatter
then we get the international form (getNumberValue will return +79201234567)
finally we format the international form using a formatter again and get +7 920 123 45-67